ai_diffusion/model.py

```diff
@@ -408,9 +410,13 @@ def handle_message(self, message: ClientMessage):
             return
 
         if message.event is ClientEvent.queued:
-            self.jobs.notify_started(job)
-            self.progress = -1
-            self.progress_changed.emit(-1)
+            if message.queue_length is not None:
+                self.queue_length = message.queue_length
+                self.queue_length_changed.emit(message.queue_length)
+            if message.queue_length is None or message.queue_length == 0:
```

Comment:
This is where the "don't start the job until it's actually running" thing moved to.

Reply:
I'd suggest to consider it started in all cases, like before (just add the queue length if available).

Reply:
I guess I don't see what it buys you. Why should we consider job started if the job is not started?